### PR TITLE
Add test plugin group

### PR DESCRIPTION
**Pull request by anchpop**
# Objective

Work towards closing #5931

## Solution

- Add a `TestPlugins` group that adds `MinimalPlugins`, `AssetPlugin`, `ScenePlugin`, `WindowPlugin`, `GilrsPlugin`, `TransformPlugin`, `HierarchyPlugin`, `DiagnosticsPlugin`, and `InputPlugin`.
- Added an example for how to use it. The example gets run in CI, which should make sure that no future change ever breaks any of the plugins in Github Actions

## Limitations

- `bevy_rapier` will not work as it depends on `RenderPlugin`, but this is fixed in #5933.
- While this is usable, `LogPlugin` cannot be added until #4934 is resolved

---

### Added

`TestPlugins` group that adds plugins suitable for use in automated tests.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]
This is a good idea, and I agree with this selection.

**Comment by mockersf** [comment]
I would rather have a feature that would enable only what is interesting in CI than a plugin group. The feature would help reducing compile time.

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
The example should live in the root level `tests` directory, which will avoid the shenanigans :) We should try and publicize that better though; perhaps we can add a note to the examples README?

File: tests/automated_tests.rs [hunk 1]
```diff
@@ -0,0 +1,48 @@
+//! This example illustrates test systems.
+fn main() {
```
Comment (alice-i-cecile) [inline tests/automated_tests.rs:2]: The entire `main` function shouldn't be necessary here.

**Comment by alice-i-cecile** [comment]
@mockersf I want your opinion on the organization here. I think it's *interesting* that we're linking to this from the examples README / including it in the cargo.toml. Unsure if it's the right choice!

**Comment by Shatur** [comment]
When I test something on CI I add only plugins I need. So I wouldn't add a specific plugin group.

**Comment by alice-i-cecile** [comment]
Closing out; I don't think this is a reliable enough solution to include at the engine level.

**Pull request closed without merging**